### ISSUE TITLE
Package co.paralleluniverse.io.serialization is duplicated in galaxy

### ISSUE THREAD

**Issue by hsestupin** [puniverse/quasar#130]
Recently I just wanted to register my own class in KryoSerializer. But I've encountered an interesting behaviour. My new class was registered in **co.paralleluniverse.io.serialization.Serialization** from quasar. But when I call **Serilization.getInstance** from my code the **Serialization** class from galaxy jar (not quasar) is used and my registrations are ignored. 

I don't even understand how this works at all - is it legal to have 2 identical classes on classpath from different jars?
And how can I register my on class to Kryo?

**Comment by pron** [comment]
Yeah, that's annoying. Java does allow listing the same classes in multiple JARs -- it takes the first one -- but that doesn't seem to be what's happening here, as both of them seem to be used, so it might be a classloader issue. I'll look into that.

**Comment by hsestupin** [comment]
Why do you need duplicate of this package in Galaxy?

**Comment by pron** [comment]
Because Galaxy doesn't depend on Quasar and vice-versa, and we were too lazy to factor this package out into its own artifact...

**Comment by pron** [comment]
Does `quasar-core` appear earlier on your classpath than galaxy (it should)? Do you use anything other than Quasar that might employ class-loading tricks?

Could you post what class-loader each copy of the `Serialization` is loaded by? If you can access both, just print the classes' `.getClassLoader()`.

**Comment by hsestupin** [comment]
I'll return to this issue not sooner than in a week.
